common/SmokeTests/SmokeTest/Update-Dependencies.ps1

```diff
@@ -31,63 +35,79 @@ function Log-Warning($message) {
     }
 }
 
-Register-PackageSource `
-    -Name $NIGHTLY_FEED_NAME `
-    -Location $NIGHTLY_FEED_URL `
-    -ProviderName Nuget `
-    -ErrorAction SilentlyContinue `
+function GetAllPackages {
+    $packages = Find-Package -Source $PACKAGE_FEED_URL -AllVersion -AllowPrereleaseVersions
+    if ($Daily) {
+        return $packages | Where-Object { $_.Version.Contains("alpha") }
+    }
+    return $packages
+}
 
-# List all packages from the source specified by $FeedName. Packages are sorted
-# ascending by version according to semver rules (e.g. 4.0.0-preview.1 comes
-# before 4.0.0) not lexicographically.
-# Packages cannot be filtered at this stage because the sleet feed to which they
-# are published does not support filtering by name.
-$allPackages = Find-Package -Source $NIGHTLY_FEED_NAME -AllVersion -AllowPrereleaseVersions
+function GetLatestPackage([array]$packageList, [string]$packageName) {
+    # This assumes that the versions coming back from Find-Package are
+    # sorted ascending. It excludes any version that is in the corresponding
+    # $PACKAGE_EXCLUSIONS entry
+    $versions = ($packageList `
+        | Where-Object { $_.Name -eq $packageName }
+        | Where-Object { `
+            -not ( $PACKAGE_EXCLUSIONS.ContainsKey($packageName) `
+            -and $PACKAGE_EXCLUSIONS[$packageName].ContainsKey($_.Version)) `
+        }
+        | Select-Object -ExpandProperty Version)
 
-$baselineVersionDate = $null;
+    if (!$versions) {
+        Write-Warning "Did not find any versions for $($packageName)"
+        return
+    }
 
-# For each PackageReferecne in the csproj, find the latest version of that
-# package from the dev feed which is not in the excluded list.
-$projectFilePath = Resolve-Path -Path $ProjectFile
-[xml]$csproj = Get-Content $ProjectFile
-$csproj |
-    Select-XML $PACKAGE_REFERENCE_XPATH |
-    Where-Object { $_.Node.HasAttribute('Version') } |
-    ForEach-Object {
-        # Resolve package version:
-        $packageName = $_.Node.Include
-
-        # This assumes that the versions coming back from Find-Package are
-        # sorted ascending. It excludes any version that is in the corresponding
-        # $PACKAGE_EXCLUSIONS entry
-        $targetVersion = ($allPackages |
-            Where-Object { $_.Name -eq $packageName } |
-            Where-Object { -not ( $PACKAGE_EXCLUSIONS.ContainsKey($packageName) -and $PACKAGE_EXCLUSIONS[$packageName].ContainsKey($_.Version)) } |
-            Select-Object -Last 1).Version
-
-        if ($targetVersion -eq $null) {
-            return
-        }
+    $sorted = [AzureEngSemanticVersion]::SortVersionStrings($versions)
+    return $sorted | Select-Object -First 1
+}
 
-        Write-Host "Setting $packageName to $targetVersion"
-        $_.Node.Version = "$targetVersion"
+function SetLatestPackageVersions([object]$csproj) {
```

Comment:
Why type object?

Reply:
If there's not something better anyway, `[object]` adds no value. Everything is an object.

Reply:
Seems from below it should be `[xml]`.

Reply:
Yeah not sure why this ended up here...